components/FileUploader.vue

```diff
@@ -0,0 +1,77 @@
+<template>
+  <v-row>
+    <v-col class="pa-0">
+      <v-file-input
+        v-model="files"
+        :multiple="multiple"
+        :label="label"
+        :accept="accept"
+        :rules="[(value) => !!value || 'The file is mandatory']"
+        color="primary"
+        chips
+        counter
+        show-size
+        @click:clear="clear()"
+      />
+    </v-col>
+  </v-row>
+  <v-row>
+    <v-col cols="auto">
+      <v-btn
+        @click="upload_files(files)"
+        color="primary"
+        :disabled="!files.length && !files_uploaded"
+        :loading="loading"
+        class="pa-2"
+      >
+        Upload file(s)</v-btn
+      >
+    </v-col>
+  </v-row>
+</template>
+
+<script setup>
+  const emit = defineEmits(["files_uploaded"])
+
+  const props = defineProps({
+    multiple: { type: Boolean, required: true },
+    accept: { type: String, required: true },
+  })
+  const { multiple, accept } = toRefs(props)
+
+  const label = multiple ? "Please select file(s)" : "Please select a file"
+  const files = ref([])
+  const loading = ref(false)
+  const files_uploaded = ref(false)
+
+  const toggle_loading = useToggle(loading)
+
+  async function upload_files() {
+    toggle_loading()
+    await upload_file(
+      { route: `tools/upload_file`, files },
+      {
+        response_function: () => {
+          files_uploaded.value = true
+          emit("files_uploaded", files.value)
+        },
+      },
+    )
+    toggle_loading()
+  }
+
+  function clear() {
+    files.value = []
+    emit("files_uploaded", files.value)
+  }
+
+  watch(files, () => {
+    files_uploaded.value = false
+  })
+</script>
+
+<style scoped>
+  .v-btn {
+    text-transform: unset !important;
```

Comment:
(if the team valid this) Do this a global scope instead of inside each component or at leat with a global CSS class to use